server/src/test/java/io/crate/integrationtests/IndexUpgraderTest.java

```diff
@@ -117,4 +123,40 @@ public void test_index_upgraders_fix_column_positions() throws Exception {
         execute("select column_name, ordinal_position from information_schema.columns where table_name = 'testing' order by 1");
         assertThat(printedTable(response.rows())).isEqualTo(expected);
     }
+
+    @TestLogging("io.crate.metadata.Functions:DEBUG")
+    @Test
+    public void test_UDFs_are_loaded_timely_for_mapping_validations_including_generated_columns_involving_UDFs_can_succeed() throws Exception {
```

Comment:
The current setup do not load javascript so I thought the only way to test is to capture the log message.

Reply:
I'd tend to have a unit test here and additionally extend the bwc tests in crate-qa.

Also avoids adding a new zip with index data and the logging hack